### PR TITLE
NAV - settings page working

### PR DESCRIPTION
**Pull request by kobryan0619**
Initial PR - adds settings page to the dashbaord

image

Work still needed: 
- Get the unit and grade to populate (currently missing).
- Remove double header (currently has "Settings" and "Edit section details"

Note that this will build on Liam's work - it is possible that if Liam merges in his PR first, this will have some conflicts.

## Links

[Ticket](https://codedotorg.atlassian.net/jira/software/projects/TEACH/boards/65?selectedIssue=TEACH-1238)
[Figma](https://www.figma.com/design/5ILfiJkPQpjskBwksMDeZl/Teacher-Dashboard?m=dev)

### CODE REVIEW

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx [hunk 1]
```diff
@@ -258,6 +258,21 @@ export default function SectionsSetUpContainer({
...
+        unitId: sections[0].unitId,
+        hasLessonExtras: sections[0].lessonExtras,
+        hasTextToSpeech: sections[0].ttsAutoplayEnabled,
+        displayName: sections[0].courseDisplayName,
+      };
+    } else {
+      return null;
+    }
+  };
+
```
Comment (kobryan0619) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:275]: @lfryemason let me know your thoughts on this... we still have this issue with the data getting passed through (a different issue, but still connected to the data).  The data expected by this component is nested - I thought of arranging the incoming data directly here.  This muddies up this component a bit (and if we go this direction, a comment would be needed), but I don't really want to mess with the data coming out of the redux store.  Thoughts?

File: apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ export default function SingleSectionSetUp({
             name="grades"
             required={true}
             options={gradeOptions}
-            values={section.grade || []}
+            values={section.grade || section.grades || []}
```
Comment (kobryan0619) [inline apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx:47]: This is another "the data is arranged differently than expected" add.

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx (hunk 1 above)
Comment (lfryemason) [reply apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:275]: Could you explain this a bit more, I'm not sure I understand why there is a difference here or what data is nested.

Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:275]: The data is structured differently when it comes directly from the back end via `getScriptData` (which is how the current/old "Edit Section" page works) vs from redux (how we are setting up the navigation bar).  The image below shows the difference.   You will notice `grade` vs. `grades` and a nested data structure vs not nested.

image

Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:275]: I will also say, I think it is OK to say "that's a later problem" and ONLY merge in a PR that shows the edit page with incomplete data and handle the data issue in a separate PR.   I threw this in here to get your ideas on it a bit and model what A solution could look like.

**Review by lfryemason — APPROVED** [review]

**Pull request merged**